detekt-gradle-plugin/src/main/kotlin/dev/detekt/gradle/plugin/internal/DetektAndroidCompilations.kt

```
@@ -0,0 +1,97 @@
+        ignoredVariants.get().contains(variant.name) ||
+            ignoredBuildTypes.get().contains(variant.buildType.name) ||
+            ignoredFlavors.get().contains(variant.flavorName)
+
+    fun linkTasks(project: Project, extension: DetektExtension) {
+        val mainTaskProvider =
+            project.tasks.register("${DetektPlugin.DETEKT_TASK_NAME}Main") {
+                it.group = "verification"
+                it.description = "EXPERIMENTAL: Run detekt analysis for production classes across " +
+                    "all variants with type resolution"
+            }
+
+        val testTaskProvider =
+            project.tasks.register("${DetektPlugin.DETEKT_TASK_NAME}Test") {
+                it.group = "verification"
+                it.description = "EXPERIMENTAL: Run detekt analysis for test classes across " +
+                    "all variants with type resolution"
+            }
+
+        val mainBaselineTaskProvider =
+            project.tasks.register("${DetektPlugin.BASELINE_TASK_NAME}Main") {
+                it.group = "verification"
+                it.description = "EXPERIMENTAL: Creates detekt baseline files for production classes across " +
+                    "all variants with type resolution"
+            }
+
+        val testBaselineTaskProvider =
+            project.tasks.register("${DetektPlugin.BASELINE_TASK_NAME}Test") {
+                it.group = "verification"
+                it.description = "EXPERIMENTAL: Creates detekt baseline files for test classes across " +
+                    "all variants with type resolution"
+            }
+
+        fun variants(extension: BaseExtension): DomainObjectSet<out BaseVariant>? = when (extension) {
+            is AppExtension -> extension.applicationVariants
+            is LibraryExtension -> extension.libraryVariants
+            is TestExtension -> extension.applicationVariants
+            else -> null
```

Is this expected? Should we just throw an exception instead so if this happen we get faster and easier reports to fix it?

---

This is a copy paste from the current logic so I'm leaving it as is - I don't think we've ever had an issue for it. I want to remove all this Android-specific logic anyway so it will go soon enough.